tools/check-rule-examples.js

```diff
@@ -129,15 +129,17 @@ async function findProblems(filename) {
                     if (comment.type !== "Block" || !/^\s*eslint(?!\S)/u.test(comment.value)) {
                         continue;
                     }
-                    const { directiveValue } = commentParser.parseDirective(comment);
-                    const parseResult = commentParser.parseJsonConfig(directiveValue);
+                    const { value } = commentParser.parseDirective(comment.value);
+                    const parseResult = commentParser.parseJSONLikeConfig(value);
                     const parseError = parseResult.error;
 
                     if (parseError) {
                         problems.push({
                             fatal: true,
                             severity: 2,
-                            message: parseError.message,
+
+                            // slice off environment-specific JSON parsing error message
+                            message: parseError.message.slice(0, parseError.message.lastIndexOf(":")),
```

Comment:
I think we should keep the whole `parseError.message` here as it's useful to see why JSON parsing failed.

If this was updated because of the failing test, we should update this line in the tests instead:

https://github.com/eslint/eslint/blob/d1f0831bac173fe3e6e81ff95c5abdbf95b02b65/tests/tools/check-rule-examples.js#L68

```diff
- .replace(/(?<=' doesn't allow this comment'):.*(?=\x1B\[0m)/u, "");
+ .replace(/(?<='doesn't allow this comment'):.*(?=\x1B\[0m)/u, "");
```

Reply:
Ah good call. :+1: